ee/tabby-webserver/src/repositories/resolve.rs

```diff
@@ -0,0 +1,159 @@
+use std::{collections::HashMap, path::PathBuf, str::FromStr};
+
+use anyhow::Result;
+use axum::{
+    body::boxed,
+    http::{header, Request, Uri},
+    response::{IntoResponse, Response},
+    Json,
+};
+use hyper::Body;
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use tabby_common::{config::Config, SourceFile, Tag};
+use tower::ServiceExt;
+use tower_http::services::ServeDir;
+use tracing::error;
+
+lazy_static! {
+    static ref META: HashMap<DatasetKey, Meta> = load_meta();
+}
+
+const DIRECTORY_MIME_TYPE: &str = "application/vnd.directory+json";
+
+#[derive(Hash, PartialEq, Eq, Debug)]
+pub struct DatasetKey {
+    local_name: String,
+    rel_path: String,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Repository {
+    name: String,
+    path: Option<String>,
+}
+
+impl Repository {
+    pub fn dataset_key(&self) -> DatasetKey {
+        DatasetKey {
+            local_name: self.name.clone(),
+            rel_path: self.path_str().to_string(),
+        }
+    }
+
+    pub fn name_str(&self) -> &str {
+        self.name.as_str()
+    }
+
+    pub fn path_str(&self) -> &str {
+        self.path.as_deref().unwrap_or("")
+    }
+}
+
+#[derive(Serialize)]
+struct ListDir {
+    entries: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Meta {
+    git_url: String,
+    filepath: String,
+    language: String,
+    max_line_length: usize,
+    avg_line_length: f32,
+    alphanum_fraction: f32,
+    tags: Vec<Tag>,
+}
+
+impl From<SourceFile> for Meta {
+    fn from(file: SourceFile) -> Self {
+        Self {
+            git_url: file.git_url,
+            filepath: file.filepath,
+            language: file.language,
+            max_line_length: file.max_line_length,
+            avg_line_length: file.avg_line_length,
+            alphanum_fraction: file.alphanum_fraction,
+            tags: file.tags,
+        }
+    }
+}
+
+/// TODO: implement auto reloading logic in future (so changes produced by tabby-scheduler command will be loaded)
+fn load_meta() -> HashMap<DatasetKey, Meta> {
+    let mut dataset = HashMap::new();
+    let repo_conf = match Config::load() {
+        Ok(config) => config
+            .repositories
+            .into_iter()
+            .map(|repo| (repo.git_url.clone(), repo))
+            .collect::<HashMap<_, _>>(),
+        Err(err) => {
+            error!("load config: {}", err);
```

Comment:
This will write an error log on for all tabby instance without a config file, which is not desired.

Reply:
Previously, `load_meta` is run in a tokio task, so logging error, imo, is necessary.

Since we load data directly now, what's the expected behavior here?

Reply:
I'll say simply ignoring it